File: TypeScript/SelectionSort.ts
const datas = [35, 80, 21, 54, 11, 45, 92, 39]
let valueMin: number

const dispArray = () => {
    for (let value of datas) {
        console.log(`${value}`)
    }
}

console.log("ソートする前の要素値の並び")
dispArray()

for (let i = 0; i < datas.length - 1; i++) {
    let indexMin: number = i
    for (let j = i + 1; j < datas.length; j++) {
        if (datas[j] < datas[indexMin]) {
            indexMin = j
        }
    }
    valueMin = datas[indexMin]
    datas[indexMin] = datas[i]
    datas[i] = valueMin
}

console.log("ソートした後の要素の並び")
dispArray()

//other method
// console.log("ソートする前の要素値の並び")
// dispArray()
// datas.sort()
// console.log("ソートする後の要素値の並び")
// dispArray()

File: Python/selection_sort.py
datas = [35, 80, 21, 54, 11, 45, 92, 39]


def disp_array():
    for value in datas:
        print("%d" % value)


print("ソートする前の要素値の並び")
disp_array()

for i in range(len(datas)-1):
    index_min = i
    for j in range(i+1, len(datas)):
        if datas[j] < datas[index_min]:
            index_min = j
    value_min = datas[index_min]
    datas[index_min] = datas[i]
    datas[i] = value_min

print("ソートした後の要素の並び")
disp_array()


# other metohd
# print("ソートする前の要素値の並び")
# disp_array()
# datas.sort()
# print("ソートした後の要素の並び")
# disp_array()
